AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/QtPyScriptCanvasNodePalette.py

```diff
@@ -33,17 +33,28 @@ def search_for_node(self, node_name: str):
         returns: a list of nodes found using the provided node_name string
         """
         self.clear_search_filter()
-        helper.wait_for_condition(lambda: self.node_palette_search_box.text() == node_name, WAIT_TIME_SEC_3)
 
         self.node_palette_search_box.setText(node_name)
-        helper.wait_for_condition(lambda: self.node_palette_search_box.text() == node_name, WAIT_TIME_SEC_3)
+        self.__validate_search_field_set(node_name)
```

Comment:
You misunderstood what I instructed here before.

This validation should happen inside a function alongside the"setText" function so we don't need to do another call.

Having this abstracted call may actually remove the need for the `clear_search_filter()` function too, or be re-written to be an abstracted call for code verbosity.

Reply:
reorganized these functions after the discussion. there is an issue with waiting for search results to be generated since the test script will outpace the filtering and raise fail reports